File: lib/models/channel.dart
enum Channel {
  stable,
  beta,
  dev,
  unknown;

  factory Channel.fromString(String value) => switch (value) {
        'stable' => Channel.stable,
        'beta' => Channel.beta,
        'dev' => Channel.dev,
        _ => Channel.unknown,
      };
}

File: src/models/channel.ts
export enum Channel {
  Stable = "stable",
  Beta = "beta",
  Dev = "dev",
}

export module Channel {
  export function parse(channel: string): Channel {
    switch (channel) {
      case "stable":
        return Channel.Stable;
      case "beta":
        return Channel.Beta;
      case "dev":
        return Channel.Dev;
      default:
        throw Error("Unsupported channel: " + channel);
    }
  }
}
